Set development mode to 'plugin' in the dev environment and allow pages to be optimized when admin is logged-in (when in plugin dev mode) (#1700)
WordPress/performance: plugins/optimization-detective/readme.txt

**Filter:** `od_url_metric_freshness_ttl` (default: 1 day in seconds)

Filters the freshness age (TTL) for a given URL Metric. The freshness TTL must be at least zero, in which it considers URL Metrics to always be stale. In practice, the value should be at least an hour. If your site content does not change frequently, you may want to increase the TTL to a week:

`
<?php
add_filter( 'od_url_metric_freshness_ttl', static function (): int {
    return WEEK_IN_SECONDS;
} );
`

> During development, this can be useful to set to zero:

Reviewer: Maybe add a brief sentence explanation on why that can be helpful?

Reviewer 2: ```suggestion
During development, this can be useful to set to zero so that you don't have to wait for new URL Metrics to be requested when engineering a new optimization:
```

`
<?php
add_filter( 'od_url_metric_freshness_ttl', static function (): int {
    return 0;
} );
`

**Filter:** `od_minimum_viewport_aspect_ratio` (default: 0.4)

Filters the minimum allowed viewport aspect ratio for URL Metrics.

> The 0.4 value is intended to accommodate the phone with the greatest known aspect ratio at 21:9 when rotated 90 degrees to 9:21 (0.429). During development you may want to set this to 0 to account Dev Tools likely being open and docked to the right:

Reviewer: Better use the phrasing like below, since we shouldn't assume people "likely" have it docked to the right:
```suggestion
The 0.4 value is intended to accommodate the phone with the greatest known aspect ratio at 21:9 when rotated 90 degrees to 9:21 (0.429). During development when you have the DevTools console open on the right, the viewport aspect ratio will be smaller than normal. In this case, you may want to set this to 0:
```

`
<?php
add_filter( 'od_minimum_viewport_aspect_ratio', static function (): int {
    return 0;
} );
`

**Filter:** `od_maximum_viewport_aspect_ratio` (default: 2.5)

Filters the maximum allowed viewport aspect ratio for URL Metrics.

The 2.5 value is intended to accommodate the phone with the greatest known aspect ratio at 21:9 (2.333).

> During development when you have the DevTools console open on the bottom, for example, the viewport aspect ratio larger than normal. In this case, you may want to increase the maximum aspect ratio:

Reviewer: Missing part of the sentence:
```suggestion
During development when you have the DevTools console open on the bottom, for example, the viewport aspect ratio will be larger than normal. In this case, you may want to increase the maximum aspect ratio:
```

`
<?php

> add_filter( 'od_maximum_viewport_aspect_ratio', function (): int {

Reviewer 3: ```suggestion
add_filter( 'od_maximum_viewport_aspect_ratio', static function (): int {
```
